compiler/rustc_query_impl/src/keys.rs

```diff
@@ -20,6 +20,12 @@ pub trait Key {
     /// In the event that a cycle occurs, if no explicit span has been
     /// given for a query with key `self`, what span should we use?
     fn default_span(&self, tcx: TyCtxt<'_>) -> Span;
+
+    /// If the key is a [`DefId`] or `DefId`--equivalent, return that `DefId`.
+    /// Otherwise, return `None`.
+    fn key_as_def_id(&self) -> Option<DefId> {
```

Comment:
I named this `key_as_def_id` to avoid conflict with other method names. (E.g., `CrateNum::as_def_id()`.) Alternatively, I could use fully-qualified syntax at the call site, which might be a good idea anyway. What do you think?

Reply:
Sounds reasonable to me